source/developers-guide/concept-cart-bundle/index.md

```diff
@@ -0,0 +1,424 @@
+        )
+    );
+
+    /** @var CartBundle\Domain\Cart\CalculatedCart $cart */
+    $cart = $facade->getCalculated();
+
+    echo "\n amount: " . $cart->getPrice()->getTotalPrice();
+    echo "\n amount net: " . $cart->getPrice()->getNetPrice();
+    echo "\n tax amount: " . $cart->getPrice()->getCalculatedTaxes()->getAmount();
+
+    /** @var CartBundle\Domain\Tax\CalculatedTax $tax */
+    foreach ($cart->getPrice()->getCalculatedTaxes() as $tax) {
+        echo "\n tax " . $tax->getTaxRate() . "% : " . $tax->getTax();
+    }
+}
+```
+
+### How to get deliveries
+Each cart contains a collection of deliveries, in case that the customer is logged in (requires a delivery address).
+This deliveries can be accessed over `getDeliveries()`. 
+```
+public function showDeliveriesAction()
+{
+    /** @var CartBundle\Infrastructure\StoreFrontFacade $facade */
+    $facade = $this->container->get('shopware_cart.store_front_facade');
+    $facade->add(
+        new CartBundle\Domain\LineItem\LineItem(
+            $number = 'SW10239',
+            CartBundle\Domain\Product\ProductProcessor::TYPE_PRODUCT,
+            $quantity = 50
+        )
+    );
+    $facade->add(
+        new CartBundle\Domain\LineItem\LineItem(
+            $number = 'SW10009',
+            CartBundle\Domain\Product\ProductProcessor::TYPE_PRODUCT,
+            $quantity = 25
+        )
+    );
+
+    /** @var CartBundle\Domain\Cart\CalculatedCart $cart */
+    $cart = $facade->getCalculated();
+
+    /** @var CartBundle\Domain\Delivery\Delivery $delivery */
+    foreach ($cart->getDeliveries() as $index => $delivery) {
+        echo "\n\n ---------- \n delivery #" . $index;
+
+        $price = $delivery->getPositions()->getPrices()->getTotalPrice();
+        echo "\n amount of delivery: " . $price->getPrice();
+        echo "\n tax amount of delivery: " . $price->getCalculatedTaxes()->getAmount();
+
+        echo "\n\n address: ";
+        echo ' ' . $delivery->getAddress()->getFirstname();
+        echo ' ' . $delivery->getAddress()->getLastname();
+        echo ' ' . $delivery->getAddress()->getStreet();
+        echo ' ' . $delivery->getAddress()->getZipcode();
+        echo ' ' . $delivery->getAddress()->getCity();
+
+        echo "\n delivery date: ";
+        echo $delivery->getDeliveryDate()->getEarliest()->format('Y-m-d');
+        echo ' - ';
+        echo $delivery->getDeliveryDate()->getLatest()->format('Y-m-d');
+
+        /** @var CartBundle\Domain\Delivery\DeliveryPosition $position */
+        foreach ($delivery->getPositions() as $i => $position) {
+            echo "\n\n position " . $i;
+            echo "\n quantity " . $position->getQuantity();
+            echo "\n price in delivery: " . $position->getPrice()->getPrice();
+        }
+
+    }
+}
+```
+
+## Architecture
+
+### Cart layers
+Inside the calculation process, the cart passes through different states. To provide a valid state for each service layer the this states are reflected in different classes:
```

Comment:
The cart passes through different states during the calculation process. In order to provide a valid state for each service layer, the states are reflected in different classes: